src/dune_rules/coq_config.ml

```diff
@@ -0,0 +1,183 @@
+open Import
+open Memo.O
+
+module Vars : sig
+  type t
+
+  val get : t -> string -> string
+
+  val get_path : t -> string -> Path.t
+
+  val get_bool : t -> ?default:bool -> string -> bool
+
+  val of_lines : string list -> (t, string) result
+
+  exception E of string
+end = struct
+  open Result.O
+
+  type t = string String.Map.t
+
+  let of_lines lines =
+    let rec loop acc = function
+      | [] -> Ok acc
+      | line :: lines -> (
+        match String.index line '=' with
+        | Some i ->
+          let x = (String.take line i, String.drop line (i + 1)) in
+          loop (x :: acc) lines
+        | None -> Error (Printf.sprintf "Unrecognized line: %S" line))
+    in
+    let* vars = loop [] lines in
+    Result.map_error (String.Map.of_list vars) ~f:(fun (var, _, _) ->
+        Printf.sprintf "Variable %S present twice." var)
+
+  exception E of string
+
+  let fail fmt = Printf.ksprintf (fun msg -> raise (E msg)) fmt
+
+  let get_opt = String.Map.find
+
+  let get t var =
+    match get_opt t var with
+    | Some s -> s
+    | None -> fail "Variable %S not found." var
+
+  let get_path t var = get t var |> Path.of_string
+
+  let get_bool t ?(default = false) var =
+    match get_opt t var with
+    | None -> default
+    | Some s -> (
+      match String.uncapitalize s with
+      | "true" | "yes" | "1" -> true
+      | _ -> default)
+end
+
+type t =
+  { version : int * int * int
+  ; version_string : string
+  ; ocaml_version_string : string
+  ; coqlib : Path.t
+  ; coqcorelib : Path.t
+  ; docdir : Path.t
+  ; ocamlfind : Path.t
+  ; camlflags : string
+  ; warn : string
+  ; hasnatdynlink : bool
+  ; coq_src_subdirs : string list
+  ; coq_native_compiler_default : string
```

Comment:
Personally I'd just put `native_compiler` and `lib`.

I'd suggest to have `Coq_config.t` and `Coq_version.t` as separate fields for now tho, as they require two calls, unless you wanna complicate the parsing big time.

Reply:
Internally we should have access to as much as possible, but we can restrict what is available in the language. The parsing shouldn't be affected by each other, but we can comment on this when I publish the new PR.

Reply:
Note that I plan to remove most of those, they are specific to coq_makefile and they should be embedded there.

Reply:
coqc -config is good, what is not good is many of these flags that are just for coq_makefile

Reply:
I agree however that much of it is useless and so we won't expose them in %{coq-config:..}.